Add SWD pinmux to stm32f1 family

alteman opened the pull request:
Add missing SWD pinmux definitions for STM32F1 MCU family

gautierg-st commented on scripts/genpinctrl/stm32f1-pinctrl-config.yaml:
```diff
@@ -256,6 +256,10 @@
   match: "^USB_OTG_FS_(?:DM)?(?:DP)?(?:SOF)?(?:ID)?(?:VBUS)?$"
   mode: input
 
+- name: JTAG PORT
+  match: ^(SYS|DEBUG)_((JTMS-)?SWDIO|(JTCK-)?SWCLK|JTDI|JTDO(-TRACESWO|-SWO)?|(NJ)?JTRST)$
+  mode: alternate
+
```
Nit: Move it just after I2S to keep alphabetical order

erwango replied:
This is done on top of the branch. You just need to rebase on top of https://github.com/zephyrproject-rtos/hal_stm32/commit/37842371f5ef0078ad32f16e5059c1df58b51892

gautierg-st reviewed (changes requested):
Could you also regenerate the pinctrl.dtsi so that the SWD pins are included in them (that also verifies that the regex is good).

alteman commented:
- Moved the section to follow I2S
- Regenerated the .dtsi files as a separate commit
- Rebased onto current main/HEAD (37842371)

On an unrelated note: mcu/STM32WBA50KEUx.xml has been deleted from STM32_open_pin_data in commit f4ec11f00e762e37ffc4020f6d4f20d225bc061d (STM32CubeMX 6.12.0), but dts/st/wba/stm32wba50keux-pinctrl.dtsi is still alive in this repo. Not sure if that's by design.
